web-ng/src/app/components/map/map.component.ts

```diff
@@ -370,6 +395,35 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     });
   }
 
+  private addPolygonToMap(project: Project, feature: PolygonFeature) {
+    const color = project.layers.get(feature.layerId)?.color;
+    const vertices: google.maps.LatLng[] = [];
+    feature.polygonVertices.map(vertex => {
+      vertices.push(new google.maps.LatLng(vertex.latitude, vertex.longitude));
+    });
+    // Construct the polygon.
+    const polygon = Polygon(
+      feature.id,
+      new google.maps.Polygon({
+        paths: vertices,
+        clickable: true,
+        strokeColor: color,
+        strokeOpacity: 1,
+        strokeWeight: normalPolygonStrokeWeight,
+        fillOpacity: 0,
+      })
+    );
+    polygon.addListener('click', () => {
+      polygon.setOptions({ strokeWeight: enlargedPolygonStrokeWeight });
+      this.panAndZoom(vertices[0]);
+      this.navigationService.selectFeature(feature.id);
+    });
+    if (this.map.googleMap !== undefined) {
```

Comment:
Is there a chance that this is called before the map is initialized? If so we should prevent this from happening, or polygons will just get dropped.

Reply:
this.map.googleMap is optional that why have added check it always called afterwards.

Reply:
This method should never be called before the map is initialized, so `this.map.googleMap!` should be used instead with no check.

Reply:
updated